test/test_encryption.py

```diff
@@ -47,169 +50,200 @@ def test_gpg_key_2(test_gpg_object: GPG, test_passphrase:str):
     test_gpg_object.delete_keys(key.fingerprint, passphrase=test_passphrase)
 
 @pytest.fixture
-def test_fingerprint_no_secret():
-    return "643BE0B0159AEFC2B9428D578533F9C66A9E7628"
+def test_pubkey_object(test_gpg_object: GPG, test_gpg_key) -> PubkeyObject:
+    current_keys = test_gpg_object.list_keys()
+    fingerprint = test_gpg_key.fingerprint
+
+    return PubkeyObject(
+        method=current_keys.key_map[fingerprint]["algo"],
+        key=fingerprint,
+        uids=current_keys.key_map[fingerprint]["uids"]
+
+    )
+
+
+@pytest.fixture
+def test_pubkey_nosecret(test_gpg_object: GPG, test_gpg_key) -> PubkeyObject:
+    return PubkeyObject(
+        method="1",
+        key="643BE0B0159AEFC2B9428D578533F9C66A9E7628",
+        uids=[""]
```

Comment:
Should we have something in the uids field to test the uids extraction function?